spa-dashboard-angular/src/app/search/search.component.ts

```diff
@@ -0,0 +1,85 @@
+import {Component, OnInit} from '@angular/core';
+
+import {Observable, Subject} from 'rxjs';
+
+import {
+    debounceTime, distinctUntilChanged, switchMap
+} from 'rxjs/operators';
+
+
+import {RoomService} from '../rooms/room.service';
+import {Lecture, Room, RoomList} from "../Container";
+import {LectureService} from "../lectures/lecture.service";
+
+@Component({
+    selector: 'app-search',
+    templateUrl: './search.component.html',
+    styleUrls: ['./search.component.css']
+})
+export class SearchComponent implements OnInit {
+    // rooms!: Observable<Room[]>;
+    // lectures!: Observable<Lecture[]>;
+    rooms: Room[] = [];
+    lectures: Lecture[] = [];
+
+    filterBy!: string;
+    filteredRooms: Room[] = [];
+    filteredLectures: Lecture[] = [];
+
+    private searchTerms = new Subject<string>();
+
+    constructor(private roomService: RoomService,
+                private lectureService: LectureService) {
+    }
+
+    // Push a search term into the observable stream.
+    search(term: string): void {
+        this.searchTerms.next(term);
+    }
+
+    ngOnInit(): void {
+
+        this.roomService.getRoomList().subscribe(rooms => {
+                this.rooms = rooms.rooms;
+                this.filteredRooms = [...this.rooms];
+            }
+        );
+        this.lectureService.getLectureList().subscribe(lecture => {
+                this.lectures = lecture.lectures;
+                this.filteredLectures = [...this.lectures]
+            }
+        );
+
+        // this.rooms = this.searchTerms.pipe(
```

Comment:
Do we need all this commented code? Can be re-added later on when new features are added in the next PR?